client.go

```diff
@@ -31,16 +31,37 @@ func Username() (string, error) {
 }
 
 // New returns a connected Client, or an error if it can't connect. The user
-// will be the user the code is running under.
+// will be the user the code is running under. If address is an empty string
+// it will try and get the namenode address from the hadoop configuration
+// files.
 func New(address string) (*Client, error) {
 	username, err := Username()
 	if err != nil {
 		return nil, err
 	}
 
+	if address == "" {
+		var nnErr error
+		address, nnErr = getNameNodeFromConf()
+		if nnErr != nil {
+			return nil, nnErr
+		}
+	}
+
 	return NewForUser(address, username)
 }
 
+// getNameNodeFromConf return a datanode from the system hadoop configuration
+func getNameNodeFromConf() (string, error) {
+	hadoopConf := LoadHadoopConf("")
+
+	namenodes, nnErr := hadoopConf.Namenodes()
+	if nnErr != nil {
+		return "", nnErr
+	}
+	return namenodes[len(namenodes)-1], nil
```

Comment:
Yeah I guess that's slightly better (but if we have multiple I'm not really sure what the behavior should be, outside of HA support).